lib/config.js

```diff
@@ -14,6 +14,17 @@ const { resolve } = require('path');
 const rechoir = require('rechoir');
 
 const fileTypes = {
+  '.esm.js': {
+    module: 'esm',
+    register(hook) {
```

Comment:
since rechoir supports arrays of different types to try and use, we could use this register object for .mjs and .es6. Do you think that would be wise?

Reply:
It will require major release. You ok with that?

Reply:
Though I would remove babel from mjs and es6. There is nothing to transpile in es6 except modules. mjs is modules specific extension.

Reply:
Isn't `.mjs` only supported through an experimental flag `--experimental-modules`? Until it comes out from behind the flag, I don't think we should remove it. With regard to removing `.es6`, would it not be useful for people using modules? I don't see the value in removing those because it would be removing value for folks. Please let me know your thoughts.

_(Sorry for the delay, lots of family stuff going on atm)_

Reply:
I mean support mjs and es6 only via esm loader instead of babel.

Reply:
I see. I think giving users the choice there is a good thing though. Being able to use babel _or_ esm is good freedom of choice.

Reply:
The choice
- .babel.js for babel
- .mjs, .es5 and .esm.js for esm

.mjs is handled in a more strict way similar to node experimental support. Babel there is too loose.

Reply:
In any case. esm is a better choice for modules transpiling. Babel is only converter. esm is a polyfill which handles edge cases.

Reply:
I would agree if this module was trying to be opinionated on that topic. But the module is not in the business of being opinionated about the module specification. Giving users a choice between babel and esm is not unreasonable. I will merge the PR as it is, but I will update to add @babel/register as a loader option for those two.

Reply:
But it's not an option. If user has esm and babel register for own needs nano will use both though user need to transform only modules.

esm is not opinionated. It follows spec. Babel can't do anything with spec. It does not know about node internals. So babel is not really safe with esm.